helix-term/src/commands/typed.rs

```diff
@@ -339,9 +339,12 @@ fn write_impl(
     let path = path.map(AsRef::as_ref);
 
     if config.insert_final_newline {
-        insert_final_newline(doc, view);
+        insert_final_newline(doc, view.id);
     }
 
+    // Save an undo checkpoint for any outstanding changes.
+    doc.append_changes_to_history(view);
```

Comment:
Ideally `Document::save_impl` would be in charge of this but currently `Document::append_changes_to_history` needs a `&mut View`. Maybe we can refactor that later and move this call into document.